src/tdastro/rand_nodes/scipy_random.py

```diff
@@ -9,15 +9,29 @@
 from tdastro.graph_state import transpose_dict_of_list
 
 
+class PDFFunctionWrapper:
+    """A class that just wraps a given PDF function.
+
+    Attributes
+    ----------
+    _pdf : function
+        The PDF function.
+    """
+
+    def __init__(self, func):
+        self._pdf = func
+        self.pdf = self._pdf
+
```

Comment:
Can we add `logpdf` here as well?

Reply:
Sure. I added this. It changes how the overloading works. We can't infer pdf vs logpdf from the function alone (at least not quickly), so I created two subclasses that handle each of these cases.